packages/notifi-react-card/lib/hooks/useNotifiSubscribe.ts

```diff
@@ -121,17 +120,24 @@ export const useNotifiSubscribe: ({
     [client.sendEmailTargetVerification],
   );
 
-  const getFirstDiscordTargetFromId = (
+  const handleMissingDiscordTarget = (
     newData: ClientData | null,
   ): string | undefined => {
+    const confirmedDiscordTarget = newData?.discordTargets.find(
+      (discordTarget) => discordTarget.isConfirmed === true,
+    );
+
+    if (confirmedDiscordTarget) {
+      setDiscordTargetData(confirmedDiscordTarget);
+      return;
+    }
+
     const firstDiscordTarget = newData?.discordTargets[0];
 
     if (firstDiscordTarget?.id) {
-      setDiscordId(firstDiscordTarget?.id);
-      return firstDiscordTarget?.id;
+      setDiscordTargetData(firstDiscordTarget);
+      return;
     }
-
-    return uuid();
   };
```

Comment:
might be necessary to reset if we don't find anything -- i.e. call `setDiscordTargetData(uncomfirmedTarget)`